Microsoft.Health.sln

```diff
@@ -66,6 +66,8 @@ Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "SchemaManager.Core.UnitTest
 EndProject
 Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Microsoft.Health.SqlServer.Tests.Integration", "test\Microsoft.Health.SqlServer.Tests.Integration\Microsoft.Health.SqlServer.Tests.Integration.csproj", "{5429C3A8-3429-43EA-BA05-E347CB4D1F61}"
 EndProject
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "Microsoft.Health.Checkpoints", "src\Microsoft.Health.Checkpoints\Microsoft.Health.Checkpoints.csproj", "{6D06503C-E79B-4F05-AFE1-C25386220895}"
```

Comment:
Question: Is this checkpoint designed as a generic solution based off the blob storage in this version? For example, we can support Event and IoT Connector?

Reply:
Yes the goal for this PR is two define the generic classes to be able to support checkpointing in the Events and IoT connector projects. I had taken the checkpointing that exists for IoT connector and basically added it here with the intent of supporting both the Events use cases and the IoT Connector use cases which are pretty similar. It should support other uses cases as well. I will eventually deprecate the IoT Connector checkpointing classes and replace them with this.